src/scippneutron/masking.py

```diff
@@ -0,0 +1,253 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+from __future__ import annotations
+
+import json
+import os
+from functools import partial
+from typing import TYPE_CHECKING
+
+import plopp as pp
+import scipp as sc
+from matplotlib.colors import to_rgb
+from plopp.core.typing import FigureLike
+
+if TYPE_CHECKING:
+    from mpltoolbox import Patch
+
+
+def _define_rect_mask(da: sc.DataArray, rect_info: dict) -> sc.Variable:
+    """
+    Function that creates a mask inside the area
+    covered by the rectangle.
+    """
+    x = rect_info["x"]
+    y = rect_info["y"]
+    xcoord = da.coords[x["dim"]]
+    ycoord = da.coords[y["dim"]]
+    return (
+        (xcoord >= x["min"])
+        & (xcoord <= x["max"])
+        & (ycoord >= y["min"])
+        & (ycoord <= y["max"])
```

Comment:
```suggestion
    return (
        ((xcoord >= x["min"])
        & (xcoord <= x["max"]))
        & ((ycoord >= y["min"])
        & (ycoord <= y["max"]))
```
I think this way, it does two operations on 1d arrays and one op on a 2d array. With you version, it is one op on 1d and two ops on 2d.